src/System.Management.Automation/engine/Modules/AnalysisCache.cs

```diff
@@ -1025,12 +1027,52 @@ private AnalysisCacheData()
 
         static AnalysisCacheData()
         {
-            s_cacheStoreLocation =
-                Environment.GetEnvironmentVariable("PSModuleAnalysisCachePath") ??
+            // If user defines a custom cache path, then use that.
+            string userDefinedCachePath = Environment.GetEnvironmentVariable("PSModuleAnalysisCachePath");
+            if (!String.IsNullOrEmpty(userDefinedCachePath))
+            {
+                s_cacheStoreLocation = userDefinedCachePath;
+            }
+
+            string cacheFileName = "ModuleAnalysisCache";
+            if (ExperimentalFeature.EnabledExperimentalFeatureNames.Count > 0)
+            {
+                // If any experimental features are enabled, we cannot use the default cache file because those
```

Comment:
Yes, my initial attempt was to use the same cache file but ignore updates from modules that has exp feature enabled. That attempt didn't go anywhere -- too many changes in module loading code and module analysis code, too risky for no value (even negative value because that makes the code hard to maintain).